gcc/testsuite/rust/compile/prelude_import.rs

```diff
@@ -0,0 +1,12 @@
+#![feature(prelude_import)]
+
+mod core {
+    mod prelude {
+        mod v1 {
```

Comment:
Shouldn't these be pub mods?

Reply:
so this is just a silly testcase for the attribute itself, and we need a valid import path so it doesn't error out. it's the same as doing something like

```rust
#[prelude_import]
use foo;

mod foo {}
```

just making sure we can parse the attribute itself :)

but yes, you are correct that in the real prelude import we need those mods to be public, and the prelude use to be public as well

Reply:
Ah ok. AFAIK currently we don't check the pub status of submodules, so I was worried something like this would later error out. That was my only concern.

Reply:
ah yeah, this is eventually going to break. I think we'll fix it once we get to this. we should definitely have one more look at our privacy checker because it's not great